src/sentry/static/sentry/app/views/projectSavedSearches.jsx

```diff
@@ -1,145 +1,115 @@
+  handleUserDefault = () => {
+    let {data, onUserDefault} = this.props;
+    onUserDefault({
+      data,
+      isUserDefault: true,
     });
-  },
-
-  handleDefault() {
-    this.handleUpdate(
-      {
-        isDefault: true,
-      },
-      this.props.onDefault
-    );
-  },
-
-  handleUserDefault() {
-    this.handleUpdate(
-      {
-        isUserDefault: true,
-      },
-      this.props.onUserDefault
-    );
-  },
+  };
 
   render() {
-    let data = this.props.data;
+    let {data, canModify} = this.props;
+
     return (
-      <tr>
-        <td>
-          <h5 style={{marginBottom: 5}}>{data.name}</h5>
+      <Row py={2} align="center">
+        <Flex flex="1" px={2} direction="column">
+          <SearchTitle>{data.name}</SearchTitle>
           <code>{data.query}</code>
-        </td>
-        <td style={{textAlign: 'center'}}>
-          <input
-            type="radio"
-            name="userDefault"
-            checked={data.isUserDefault}
-            onChange={this.handleUserDefault}
-          />
-        </td>
-        {this.props.access.has('project:write') && (
-          <td style={{textAlign: 'center'}}>
+        </Flex>
+        <Flex flex="1">
+          <InputColumn>
             <input
               type="radio"
-              name="default"
-              checked={data.isDefault}
-              onChange={this.handleDefault}
+              name="userDefault"
+              checked={data.isUserDefault}
+              onChange={this.handleUserDefault}
             />
-          </td>
-        )}
-        {this.props.access.has('project:write') && (
-          <td style={{textAlign: 'right'}}>
-            <Confirm
-              message={t('Are you sure you want to remove this?')}
-              onConfirm={this.handleRemove}
-              disabled={this.state.loading}
-            >
-              <a className="btn btn-sm btn-default">
-                <span className="icon icon-trash" /> &nbsp;{t('Remove')}
-              </a>
-            </Confirm>
-          </td>
-        )}
-      </tr>
+          </InputColumn>
+
+          {canModify && (
+            <InputColumn>
+              <input
+                type="radio"
+                name="default"
+                checked={data.isDefault}
+                onChange={this.handleDefault}
+              />
+            </InputColumn>
+          )}
+
+          {canModify && (
+            <InputColumn>
+              <Confirm
+                message={t('Are you sure you want to remove this?')}
+                onConfirm={this.handleRemove}
+              >
+                <Button size="small">
+                  <span className="icon icon-trash" />
+                </Button>
+              </Confirm>
+            </InputColumn>
+          )}
+        </Flex>
+      </Row>
     );
-  },
-});
+  }
+}
 
 const ProjectSavedSearches = createReactClass({
   displayName: 'ProjectSavedSearches',
-  mixins: [ApiMixin, OrganizationState],
+  contextTypes: {
```

Comment:
Removed `OrganizationState` in favor of context because latter is easier to test